### PR TITLE
[Product] Product search for customer

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets |
| License         | MIT

Maybe it doesn't look great, but at least it works 😄  🐫 

zrzut ekranu 2016-10-25 o 11 44 18

zrzut ekranu 2016-10-25 o 11 40 47

### CODE REVIEW

File: src/Sylius/Component/Core/Repository/ProductRepositoryInterface.php [hunk 1]
```diff
@@ -33,10 +33,12 @@ public function createQueryBuilderWithLocaleCodeAndTaxonId($localeCode, $taxonId
     /**
      * @param string $code
      * @param ChannelInterface $channel
+     * @param string $locale
      *
      * @return QueryBuilder
      */
-    public function createQueryBuilderForEnabledByTaxonCodeAndChannel($code, ChannelInterface $channel);
+    public function createQueryBuilderForEnabledByTaxonCodeAndChannel($code, ChannelInterface $channel, $locale);
```
Comment (pamil) [inline src/Sylius/Component/Core/Repository/ProductRepositoryInterface.php]: `$locale` -> `$localeCode`?

Comment (Zales0123) [inline src/Sylius/Component/Core/Repository/ProductRepositoryInterface.php]: It's not consistent, but in most repository methods we use ``$locale``. I think it's clear enough, as in code it almost everywhere a locale _code_, not _name_, but we should obviously keep a consistency with this.

File: src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php [hunk 2]
```diff
@@ -51,14 +51,18 @@ public function createQueryBuilderWithLocaleCodeAndTaxonId($localeCode, $taxonId
     /**
      * {@inheritdoc}
      */
-    public function createQueryBuilderForEnabledByTaxonCodeAndChannel($code, ChannelInterface $channel)
+    public function createQueryBuilderForEnabledByTaxonCodeAndChannel($code, ChannelInterface $channel, $locale)
```
Comment (bretto36) [inline src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php]: Should this have AndLocale in the function name?

File: src/Sylius/Bundle/ShopBundle/Resources/config/grids/product.yml [hunk 3]
```diff
@@ -13,6 +13,7 @@ sylius_grid:
                         arguments:
```
Comment (bretto36) [inline src/Sylius/Bundle/ShopBundle/Resources/config/grids/product.yml]: change method above to have AndLocale in the method name

File: src/Sylius/Bundle/ShopBundle/Resources/views/Grid/Filter/string.html.twig [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{% form_theme form 'SyliusUiBundle:Form:theme.html.twig' %}
+
+{{ form_widget(form.type, { 'value': 'contains', 'attr': { 'style': 'display: none;' } }) }}
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Grid/Filter/string.html.twig]: Twig CS! :) No spaces around { and '.

**Comment by michalmarcinkowski** [comment]
Thank you Mateusz!

**Comment by gabiudrescu** [comment]
any ETA about when I could test this on http://demo.sylius.org/ ?

**Comment by okwinza** [comment]
any ETA on elasicsearch support yet, if any?

**Comment by gabiudrescu** [comment]
@okwinza maybe this helps? https://github.com/Sylius/Sylius/pull/5756

**Pull request merged**